runtime/common/CMakeLists.txt

```diff
@@ -93,13 +93,11 @@ add_library(cudaq-mlir-runtime
 )
 set_property(GLOBAL APPEND PROPERTY CUDAQ_RUNTIME_LIBS cudaq-mlir-runtime)
 
-# Note: JIT.cpp contains throw statements. Should RTTI be enabled?
 set_source_files_properties(
     Environment.cpp
     JIT.cpp
     Logger.cpp
     RuntimeMLIR.cpp
-  PROPERTIES COMPILE_FLAGS -fno-rtti
```

Comment:
@amccaskey Do you recall why this had RTTI disabled?

Reply:
i believe this was to get the build working against LLVM/MLIR which by default turns off rtti in favor of its hand-rolled version

Reply:
@amccaskey since the build works with rtti enabled now, is this change ok, or are there some hidden problems that we might be concerned about?

Reply:
all tests pass so I think we are good!